Windows nightly: Thrift 0.13.0 build fails with Boost 1.83.0 due to missing NOMINMAX [HZ-5424]

## Summary

The nightly Windows CI build fails when Thrift 0.13.0 is compiled against Boost 1.83.0. The error originates in `boost/locale/util/string.hpp` which was introduced in Boost 1.83.0 and calls `std::numeric_limits<char>::max()` inside a `constexpr` function.

**Failing job:** https://github.com/hazelcast/hazelcast-cpp-client/actions/runs/26226738709/job/77175714903  
**Configuration:** Windows · Boost 1.83.0 · Debug · noSSL

## Error

```
C:\Boost\include\boost-1_83\boost\locale\util\string.hpp(41,66): error C2589: '(': illegal token on right side of '::'
C:\Boost\include\boost-1_83\boost\locale\util\string.hpp(41,66): error C2059: syntax error: ')'
C:\Boost\include\boost-1_83\boost\locale\util\string.hpp(39,20): error C3615: constexpr function 'boost::locale::util::to_char' cannot result in a constant expression
  (compiling source file '../../../lib/cpp/src/thrift/protocol/TJSONProtocol.cpp')
```

## Root Cause

On Windows, `windows.h` defines `min` and `max` as macros unless `NOMINMAX` is defined beforehand. Boost 1.83.0 added `boost/locale/util/string.hpp` which contains:

```cpp
constexpr char to_char(int c) noexcept {
    return (c >= 0 && c <= std::numeric_limits<char>::max()) ? ...;
}
```

Because Thrift 0.13.0 does not define `NOMINMAX` globally, the `max` macro corrupts this call at the preprocessor level, causing the C2589/C2059/C3615 errors.

## Fix

Pass `/DNOMINMAX` via `CMAKE_CXX_FLAGS` when configuring the Thrift CMake build in the Windows CI action. This is a safe, idiomatic Windows fix that defines `NOMINMAX` before any header is processed and does not affect other configurations.

--- a/.github/actions/build-test/windows/action.yml
+++ b/.github/actions/build-test/windows/action.yml
@@ -118,7 +118,8 @@ runs:
                   -DBUILD_COMPILER=OFF -DBUILD_TESTING=OFF -DBUILD_TUTORIALS=OFF -DBUILD_LIBRARIES=ON `
                   -DBUILD_CPP=ON -DBUILD_AS3=OFF -DBUILD_C_GLIB=OFF -DBUILD_JAVA=OFF -DBUILD_PYTHON=OFF `
                   -DBUILD_HASKELL=OFF -DWITH_OPENSSL=OFF -DWITH_LIBEVENT=OFF -DWITH_ZLIB=OFF `
-                  -DWITH_QT5=OFF -DBoost_INCLUDE_DIR=${{ inputs.BOOST_INCLUDE_FOLDER }} -DCMAKE_INSTALL_PREFIX=C:\Thrift
+                  -DWITH_QT5=OFF -DBoost_INCLUDE_DIR=${{ inputs.BOOST_INCLUDE_FOLDER }} -DCMAKE_INSTALL_PREFIX=C:\Thrift `
+                  "-DCMAKE_CXX_FLAGS=/DNOMINMAX"
         cmake --build . --target install --config ${{ inputs.BUILD_TYPE }}
         mkdir C:\Thrift\bin
         cd ../..

